docs/guides/settings.rst

```diff
@@ -60,6 +64,10 @@ where:
 - ``$user_guid`` Is the user you want to retrieve these for (defaults to the currently logged in user)
 - ``$plugin_name`` Is the name of the plugin (detected if run from within a plugin)
 
+.. note::
+
+   Settings stored as an array will be returned as the serialized value.  Developers are expected to unserialize such settings on retrieval.
```

Comment:
I agree it's a bit of a discrepancy in that they have to manually unserialize, but it's what they're doing anyway.  This is just making the input part a little DRY-er.
I don't think we can get a consistent array-in/array-out behavior in a BC fashion.